### ISSUE TITLE
uncaught exception: TypeError: (err.message || "").match is not a function

### ISSUE THREAD

**Issue by bonesoul** [opbeat/opbeat-node#130]
getting with v4.11.3 
```
uncaught exception: TypeError: (err.message || "").match is not a function
    at setException (/var/www/firsat.io/node_modules/opbeat/lib/parsers.js:149:35)
```

**Comment by watson** [comment]
Thanks for reporting. Please post the complete stack trace if possible.

Are you calling `opbeat.captureError` manually and in that case what type of object are you passing in as the first argument?

**Comment by bonesoul** [comment]
nop i'm not making any manual calls;

full stack trace:

```
error uncaught exception: TypeError: (err.message || "").match is not a function
    at setException (/var/www/firsat.io/node_modules/opbeat/lib/parsers.js:149:35)
    at /var/www/firsat.io/node_modules/opbeat/lib/parsers.js:33:5
    at /var/www/firsat.io/node_modules/stackman/index.js:71:9
    at f (/var/www/firsat.io/node_modules/once/once.js:25:25)
    at /var/www/firsat.io/node_modules/after-all/index.js:35:11
    at instrumented (/var/www/firsat.io/node_modules/opbeat/lib/instrumentation/index.js:102:27)
    at /root/.nvm/versions/node/v7.7.2/lib/node_modules/pm2/node_modules/async-listener/glue.js:188:31
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickDomainCallback [as _tickCallback] (internal/process/next_tick.js:128:9)

```

**Comment by watson** [comment]
Weird. I'll make a patch for this, but I'd also really like to know why `err.message` isn't a string in this case. Do you know if anything in your code-base is overwriting the `message` property on error objects?

**Comment by bonesoul** [comment]
nah; here is my usage;

```
  .then(function () {
    return new Promise(function (resolve) {
      winston.info('[MONITOR] opbeat monitoring: %s.', config.get('monitoring.opbeat.enabled') ? 'enabled' : 'disabled');
      if (!config.get('monitoring.opbeat.enabled')) return resolve(); // only initialize the opbeat support if enabled.

      // initialize opbeat
      require('opbeat').start({
        appId: config.get('monitoring.opbeat.appId'),
        organizationId: config.get('monitoring.opbeat.organizationId'),
        secretToken: config.get('monitoring.opbeat.secretToken'),
        logBody: true,
        captureExceptions: true,
        captureTraceStackTraces: true
      });

      return resolve();
    });
  })

  if (config.get('monitoring.opbeat.enabled')) app.use(monitor.opbeat()); // initialize the opbeat support if enabled.

const opbeat = require('opbeat');

exports.opbeat = function () {
  return opbeat.middleware.express();
};
```

**Comment by watson** [comment]
It's fixed and shipped as v4.11.4 - Once again, thanks for reporting this 😃

When you run your app with the new version of Opbeat and it actually manages to log the error it's trying to log, then I'll love to hear if you can see more details of what the `err.message` property was actually set to. I've never seen it behave like this before...
